pandas/core/dtypes/inference.py

```diff
@@ -262,7 +263,7 @@ def is_list_like(obj):
     False
     """
 
-    return (hasattr(obj, '__iter__') and
+    return (hasattr(obj, '__iter__') and isinstance(obj, Iterable) and
```

Comment:
this should be an ``or``

Reply:
I see your point, but that wont fix the issue. What about only check `isinstance(obj, Iterable)`? I'm asking this because I see another function in `inference.py` called `_iterable_not_string`, it checks against `collections.Iterable` rather check for `__iter__` attribute.

Reply:
you can try but this is used pretty much everywhere and you will break things. Technically a ``type`` satisfies this condition I think, but we DO want to exlude it. you could also add it to the ``not isinstance(obj, (string_types, type))``. might work.

Reply:
I've checked,  `type` actually doesn't satisfy the condition. Also all the tests have passed after the change.